osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -22,6 +22,11 @@ public abstract class Mod
         public virtual FontAwesome Icon => FontAwesome.fa_question;
 
         /// <summary>
+        /// The type of this mod.
+        /// </summary>
+        public virtual ModType Type => ModType.Special;

```

Comment:
Now that this is in the mod class, how about fixing the methods which previously had this hardcoded?

ie.
`public override IEnumerable<Mod> GetModsFor(ModType type)`